relay-event-schema/src/protocol/ourlog.rs

```diff
@@ -0,0 +1,253 @@
+use relay_protocol::{
+    Annotated, Empty, Error, FromValue, IntoValue, Object, SkipSerialization, Value,
+};
+
+use serde::ser::SerializeMap;
+
+use crate::processor::ProcessValue;
+use crate::protocol::{SpanId, TraceId};
+
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[metastructure(process_func = "process_ourlog", value_type = "OurLog")]
+pub struct OurLog {
+    /// Time when the event occurred.
+    #[metastructure(required = true, trim = false)]
+    pub timestamp_nanos: Annotated<u64>,
+
+    /// Time when the event was observed.
+    #[metastructure(required = true, trim = false)]
+    pub observed_timestamp_nanos: Annotated<u64>,
+
+    /// The ID of the trace the log belongs to.
+    #[metastructure(required = false, trim = false)]
+    pub trace_id: Annotated<TraceId>,
+    /// The Span id.
+    ///
+    #[metastructure(required = false, trim = false)]
+    pub span_id: Annotated<SpanId>,
+
+    /// Trace flag bitfield.
+    #[metastructure(required = false)]
+    pub trace_flags: Annotated<f64>,
+
+    /// This is the original string representation of the severity as it is known at the source
+    #[metastructure(required = false, max_chars = 32, pii = "true", trim = false)]
+    pub severity_text: Annotated<String>,
+
+    /// Numerical representation of the severity level
+    #[metastructure(required = false)]
+    pub severity_number: Annotated<i64>,
+
+    /// Log body.
+    #[metastructure(required = true, pii = "true", trim = false)]
+    pub body: Annotated<String>,
+
+    /// Arbitrary attributes on a log.
+    #[metastructure(pii = "true", trim = false)]
+    pub attributes: Annotated<Object<AttributeValue>>,
+
+    /// Additional arbitrary fields for forwards compatibility.
+    #[metastructure(additional_properties, retain = true, pii = "maybe", trim = false)]
+    pub other: Object<Value>,
+}
+
+#[derive(Debug, Clone, PartialEq, ProcessValue)]
+pub enum AttributeValue {
```

Comment:
Moving it over to `enum` caused the traits to not derive easily, or at least I couldn't figure out a simple way of doing it. Any suggestions to simplify this or not have to have the `impl` below are welcome.

Reply:
I don't have a better idea to be honest.